### PR TITLE
DDP-7628/REFACTORING

### PR DESCRIPTION
**Pull request by GiCharkviani**
https://broadinstitute.atlassian.net/browse/DDP-7628 And REFACTORING!

Notes:
1. Please, don't pay attention how HTML files look like, as it is temporary and will be updated and optimized when UI template is provided.
2. All is tested and working, but still, further optimizations and improvements will be done slowly, hence I would take any suggestion for future improvements.

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html [hunk 1]
```diff
@@ -1,131 +1,141 @@
...
+             let-stableId="stableId"
+             let-question="question"
+             let-answer="questionTypes.answer"
+             let-composite="questionTypes.composite"
+             let-picklist="questionTypes.picklist"
+             let-matrix="questionTypes.matrix"
+>
+  <ng-container *ngIf="type === 'TEXT'">
+    <div *ngIf="answer">
+      <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html:51]: Maybe we can have this color in a css file so we can reuse it again?

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html [hunk 2]
```diff
@@ -1,131 +1,141 @@
...
+      <li *ngFor="let mat of matrix"><b>{{mat.question}}:</b> {{mat.answer}}</li>
+    </ul>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'DATE'">
+    <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+    <p><b>{{ answer }}</b></p>
+  </ng-container>
+
+</ng-template>
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html:141]: The overall file looks good, I'm just wondering, shouldn't we get the "nice" text from the QuestionDefinition for all the texts that we are displaying?

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html (hunk 1 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html:51]: Yes, almost all that you see in the HTML file are temporary. There won't be inline CSS when we have UI (or I will do it on my own when I get time).

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html (hunk 2 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html:141]: As I have already said, almost everything will be changed and will look much better. For this, I need a UI template or just some free time to do it on my own.

Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html:141]: No that is fine, I saw that you are using the ActivityDefinition with the buildActivity method, right?

**Review by pegahtah — APPROVED** [review]

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html (hunk 2 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html:141]: Yes, in order to build survey data (thinking about changing some variable and method names, to be more clear :) )

**Pull request merged**